testing/ldc/APKBUILD

```diff
@@ -10,73 +10,85 @@ url="https://github.com/ldc-developers/ldc"
 # Once we can use GDC for bootstrapping, this can be "all"
 arch="x86_64"
 license="BSD-3-Clause AND BSL-1.0 AND (Artistic-1.0 OR GPL-2.0-or-later) AND NCSA AND MIT"
-depends="libexecinfo"
-makedepends="chrpath cmake llvm5-libs libexecinfo-static"
+depends="binutils-gold libexecinfo"
+makedepends="chrpath cmake dlang-compiler llvm9-dev llvm9-static curl-dev libedit-dev libexecinfo-static zlib-dev"
 # A user might want to install the '-runtime' subpackage when they have
 # a dynamically-linked D program.
 subpackages="$pkgname-runtime $pkgname-doc $pkgname-static"
 provides="dlang-compiler"
-source="https://github.com/ldc-developers/ldc/releases/download/v$pkgver/ldc2-$pkgver-alpine-linux-x86_64.tar.xz
+source="https://github.com/ldc-developers/ldc/releases/download/v$pkgver/ldc-$pkgver-src.tar.gz
 		ldc2.package.conf"
 builddir="$srcdir/"
 
 build() {
-	rootdir="$srcdir/ldc2-$pkgver-alpine-linux-x86_64/"
+	rootdir="$srcdir/ldc-$pkgver-src/"
 
-	# Unfortunately the upstream-provided version has the builder's /home/ in its rpath - strip it
-	chrpath -d "$rootdir/bin/ldc2"
-	chrpath -d "$rootdir/bin/ldmd2"
-	# This exe is not installed, just used in build, but do it for completeness
-	chrpath -d "$rootdir/bin/ldc-build-runtime"
-
-	# Now build the runtime - the compiler itself is already built by upstream
-	"$rootdir"/bin/ldc-build-runtime --buildDir="$builddir/ldc-runtime-build/"
+	cd "$rootdir"
+	if [ "$CBUILD" != "$CHOST" ]; then
+		CMAKE_CROSSOPTS="-DCMAKE_SYSTEM_NAME=Linux -DCMAKE_HOST_SYSTEM_NAME=Linux"
+	fi
+	cmake \
+		-DCMAKE_INSTALL_PREFIX=/usr \
+		-DCMAKE_INSTALL_LIBDIR=lib \
+		-DBUILD_SHARED_LIBS=True \
+		-DCMAKE_BUILD_TYPE=Release \
+		-DCMAKE_CXX_FLAGS="$CXXFLAGS" \
+		-DCMAKE_C_FLAGS="$CFLAGS" \
+		$CMAKE_CROSSOPTS "$rootdir"
+	make
 
 	# CMake added the rpaths to the shared libs - strip them
-	chrpath -d "$builddir"/ldc-runtime-build/lib/*.so
+	chrpath -d "$rootdir"/lib/*.so*
 	:
 }
 
 check() {
-	rootdir="$srcdir/ldc2-$pkgver-alpine-linux-x86_64/"
-	libroot="$builddir/ldc-runtime-build/lib/"
+	rootdir="$srcdir/ldc-$pkgver-src/"
 
-	echo 'import std.stdio; void main () { try throw new Exception(null); catch (Exception) writeln("Hello World!"); }' > hello_world.d
-	if [ "$($rootdir/bin/ldmd2 -conf='' -I$rootdir/import/ldc -I$rootdir/import/ -L-L$libroot -defaultlib=phobos2-ldc,druntime-ldc -L-lexecinfo -L-rpath=$libroot -run hello_world.d)" != "Hello World!" ]; then
-		return 1
-	fi
+    cd "$rootdir"
+    make test
 	:
```

Comment:
I don't think this is required anymore

Reply:
You mean the column (`:`) right ? Actually I was wondering what it was for... Does it have a name ? Google hasn't been helpful.

Reply:
Not quite sure what it's name, but you use it when you have an empty function, since this would be a syntax error:

```
check() {

}
```

whereas

```
check() {
    : 
}
```

isn't. You could put a `true` in there too, I suppose.